www/js/diary/infinite_scroll_list.js

```diff
@@ -75,14 +76,16 @@ angular.module('emission.main.diary.infscrolllist',['ui-leaflet',
 
   $scope.getCardHeight = function(entry) {
     let height = 0;
-    if (entry.start_ts) { // entry is a trip
+    if (entry.enter_ts) { // entry is a place
+      height = 106;
+    } else if (!entry.locations) { // entry is untracked time
+      height = 130;
+    } else if (entry.start_ts) { // entry is a trip
```

Comment:
I think it would be better to put a type into the entry instead of using these indirect inferences.  It will ensure that we have flexibility for different types of places and trips down the road, and will ensure that there is a central location where we set the types.

Having such indirect inferences sprinkled through the code is going to make it super hard to change in the future.

Reply:
This is true for the `isPlace` checks as well. The data model on the server is already typed. Why don't we just propagate that type throughout instead of inferring place vs. trip in every location where we need to handle them.

Reply:
I definitely agree. I'll make sure every triplike or place object on the label screen has a 'key' describing what data type it comes from.
Currently when composite trips are unpacked / "flattened",  the metadata is sort of just thrown away. so I'll make sure to grab the key beforehand